### DISCUSSION TITLE
Keyboard operation on new Cinnamon dialog.

### DISCUSSION THREAD

**Discussion by reasonset** [linuxmint/discussions#694] [Ideas]
In the past, Cinnamon dialogs could be operated with the keyboard, but with the new dialogs, those without text input can no longer be operated with the keyboard.
For example, shutdown and logout dialogs. In the past, these could be operated using the cursor keys, enter key, and alt key, but we find it very inconvenient that these are no longer possible.
Is it possible to make the new dialogs keyboard-operated?

**Comment by JosephMcc** [comment]
This should be fixed on git master and in the next Cinnamon update.

**Reply by reasonset** [reply]
Thank you!

**Comment by fnetX** [comment]
Hi there, I just received the 6.4.6 shipped with LMDE 6. I am also affected by this and somewhat disappointed that user research / design (if there was any) did not consider keyboard navigation from the start, especially considering the accessibility aspect (I am personally only speaking as an annoyed user who thinks that keyboard navigation is crucial for an efficient interaction with a machine, but I think that everyone who cannot (conveniently) use a mouse should always be offered an adequate replacement).

I am used to quickly interacting interfaces using the alt key. For example, I suspend my desktop frequently using a keyboard shortcut and then using <kbd>alt</kbd>+<kbd>u</kbd> to suspend. Same for many other dialogs. However, this is now broken, and <kbd>left key</kbd><kbd>left key</kbd><kbd>left key</kbd><kbd>enter</kbd> is not really an efficient workaround.

Thank you for the consideration.

**Reply by fnetX** [reply]
As a funny anecdote: My laptop broke recently, and to get back up I installed the nvme ssd into an USB adapter and booted over USB. This worked fine on USB3, but I had another workstation where I booted over USB2, which was generally fine but …

I am so used to the timely keyboard navigation that I even had a race condition against the machine. Usually, the dialog would pop up so quickly, that me hitting alt+u immediately would suspend the machine as I expect. However, via USB2, the dialog was delayed so that I was doing that sequence faster than the machine could bear, which also interrupted my workflow.

I know that it is very difficult to satisfy everyone who uses keyboard navigation, because workflows can become muscle memory. However, it demonstrates that keyboard navigation can become very efficient over time, and breaking this workflow is usually a serious issue.

It is possible to get used to other shortcuts, but not considering keyboard navigation at all is a big issue in my eyes.

**Comment by reasonset** [comment]
I am also very pleased to confirm that the dialog can now be operated with the keyboard. Thank you very much.

But even for me this is not enough.
I used to use Alt+R to reboot, but this operation is no longer available.
I would also like to see the combination with the Alt key restored if possible, as the cursor keys are not easy to press on a 60% keyboard or compact keyboard.

**Reply by Loufute** [reply]
Hi,
I just updated to Mint 22.1, and thus latest Cinnamon, and I would like to join this request. I was used to use alt+r as well to reboot (for instance), which does not work anymore. Also, pushing on "alt" does not underline letter of the shortcut. I would be very happy if both could be back.
Best,
Loufute

**Reply by cplittlfield** [reply]
There is a programmable shortcut to shut down the system.  The default shortcut is Ctrl + Alt + End, but there is nothing to stop you from adding a key combination you would prefer over the default.  You can have up to three key combinations per action.

Forgot:  System Settings > Keyboard > Shortcuts tab.  Click on System in the left pane, and you will see "Shut Down" as the second option in the top right pane.  Select it and follow the instructions.

**Reply by Loufute** [reply]
Dear cplittlfield,

I appreciate the time you took to explain this. However, I am already aware of personnalisable shortcuts, which I avoid to shut down or restart the computer (I remove Ctrl + Alt + End), as it creates unwanted surprises.
Having shortcuts on the shutdown dialog allows a two-step action, which corresponds better to my needs.

Best,

Loufute

**Reply by cplittlfield** [reply]
I understand, and I don't know why they failed to add the shortcuts to the dialogues created with Clutter.  Perhaps someone forgot.  It is not a problem for me when I actually want to shut down, since the Shutdown option is the default (just hit Enter); the problem comes when I want to Restart, since I have to take my hands off the keyboard, reach for the pen, find the cursor, and get it onto the correct button, all before the timer runs out and the system shuts down anyway.  I have to set a good long timer delay in the Cinnamon settings.

**Reply by JosephMcc** [reply]
The shortcuts were actually an oversight. This however is being a bit dramatic:

>  the problem comes when I want to Restart, since I have to take my hands off the keyboard, reach for the pen, find the cursor, and get it onto the correct button, all before the timer runs out and the system shuts down anyway. I have to set a good long timer delay in the Cinnamon settings.

That Tab and arrow keys allow navigation in the dialog. So I understand if not everyone loves the new design, but lets not be silly about it.

**Reply by cplittlfield** [reply]
> That Tab and arrow keys allow navigation in the dialog.

Really?  They don't for me--or if they do, there is no visual indicator of which option is selected, so the keys might just as well not be working.  That's why I've been making a deal about the shortcuts.

**Reply by JosephMcc** [reply]
They work fine. If there is no indication, then your theme isn't properly supporting the button styles. Try with the default cinnamon theme. ESC also cancels any of these dialogs.

**Reply by cplittlfield** [reply]
> They work fine. If there is no indication, then your theme isn't properly supporting the button styles. Try with the default cinnamon theme.

Which would be?  I am currently using Mint-X as the button theme.  What should I be using instead?  Mint-Y isn't any better.

If there is a theme that makes window and button borders clearly visible to these old eyes, I would gladly use it.

> ESC also cancels any of these dialogs.

And thank God for that!

**Reply by JosephMcc** [reply]
Cinnamon comes with a default theme called "Cinnamon" you can set as the desktop theme. This is essentially the reference theme that gets updated for any new things added to Cinnamon.

**Reply by cplittlfield** [reply]
Well, that's weird!  I have all kinds of themes listed when I go to System Settings > Themes > Themes and click on one of the buttons, but the long lists that appear have no theme named "Cinnamon."  The Add/Remove tab doesn't show such a thing, either, unless you are referring to one of the Cinnamox variants.  I don't see any themes named "Cinnamon" in the Software Manager either.  What am I doing wrong?

**Reply by reasonset** [reply]
This may seem trivial, and it may be a matter of familiarity, but I have been used to Alt+R to reboot and Alt+U to suspend, and I do not find it very convenient to use the cursor keys.

This is especially true when I am using a compact keyboard that does not have separate cursor keys.

Perhaps it is because my fingers used to memorize the operation and perform it without looking at the screen, but I feel that using cursor keys is troublesome because I have to visualize which button has the focus.

**Reply by JosephMcc** [reply]
@reasonset Like I said, those shortcuts were overlooked. Can look at how hard they would be to add back.

**Reply by reasonset** [reply]
@JosephMcc 
Perhaps only a small percentage of people crave that shortcut, and if it cannot be easily added back, I think I will have to give up on it.

However, it is true that it is a feature that I would like to have, so I would be happy if you could keep it in mind.

**Reply by ghost** [reply]
@reasonset You can make keyboard shortcuts, in the keyboard settings menu under "short cuts" and use the search bar to find the "shutdown" key combo, which is currently for me at least is ctrl+alt+end, and just change it to what you want.

**Reply by reasonset** [reply]
@fosslover306 It's not I want.
I want key operation at the context, not a global shortcut.

**Reply by ghost** [reply]
@reasonset I was referring to your alt+r and alt+u not being the default comment, not the keyboard navigation with the new dialogs comment which I do agree needs to come back if it hasn't yet, but that feature is back already on LMDE6.

Basically the current shortcut's are fine for global hotkeys, and if you are using the gui dialog you clearly don't intend to use a shortcut anyway because that would be completely redundant.

Arrow keys are good enough if you just want to select an option from the system dialog, and as long as you can press the "ENTER" key to execute your choice then it's perfectly navigational with a keyboard.

And I got to ask what compact keyboard does not have dedicated arrow keys? (note: not cursor keys)

**Reply by reasonset** [reply]
@fosslover306 
I don't know what you mean by arrow keys that are not cursor keys, but generally 60% keyboards do not have arrow keys.

I run about 30 machines, so setting up global shortcuts would be quite a challenge, and I would have to make sure there are no conflicts with other software, and still set up consistent shortcuts for all environments.

In addition, being able to reboot or shutdown in a global context could be an accident; I often use programs that need to run for a month or so, and an unexpected shutdown would be a loss.

And what I need is not to be able to perform shutdowns with shortcuts, but to be able to use key combinations for navigation in dialogs. Not all dialogs can be assigned with global shortcuts, and not all problems can be solved with shortcuts.

In a nutshell, you and I have different environments.
I need it and you don't. That's all.

**Reply by ghost** [reply]
These are arrow keys: https://en.wikipedia.org/wiki/Arrow_keys
Cursor keys are usually the ones that can also be number keys depending on if you are using numlock or not.

I'm just curious how one ends up with 30 keyboards missing arrow keys, even the 60% keyboards on amazon have arrow keys, granted a lot less then I expected but they exist for the same price as arrow key-less options with a QWERTY layout.

That being said how do you even make use of a keyboard that crappy in the first place with a GUI interface, considering the fact that it makes file management nearly impossible without a mouse never mind any other system menu, or even the command line.

Considering my daily file management activities if I was stuck with a keyboard like that, Well let just say hell would be preferable as far as i'm concerned.

To be clear i'm not judging, criticizing or making fun of you I'm just completely bewildered that such a situation exists and why would such a situation even be considered reasonable to create.

I can see your point about the shortcut tho.

**Reply by reasonset** [reply]
@fosslover306 
I have four 60% keyboards: the Razer HUNTSMAN MINI (JP), DUCKY ONE 3 Mini (US), iQunix EZ60 (US), and Wooting 60 (US). None of them have independent arrow keys.

The 60% keyboard is for gaming and I love TKL, but the reason I use the 60% keyboard is because of the size of my desk and portability.
When using multiple machines at the same time, there are times when I really need more than one keyboard, and with the TKL you can only have two on my desk, but with the 60% keyboard I can have three.
When I take the 60% keyboard to work on office because I don't want to carry the TKL.

Such 60% keyboards do not have independent arrow keys, but they can use them in combination with Fn. However, it is quite stressful.

Of course, I use TKL on my main PC, and I use SSH as much as possible.
And by the way, all but the HUNTSMAN mini were gifts, so they don't reflect my preferences.

**Reply by JosephMcc** [reply]
@reasonset You don't need the arrow keys. The tab key will also cycle through the buttons.

**Reply by ghost** [reply]
@JosephMcc That's cool, I did not know that was an option.

**Reply by Loufute** [reply]
> @reasonset You don't need the arrow keys. The tab key will also cycle through the buttons.

Just to add on that (in case that helps someone), as often when there is TAB, SHIFT+TAB also works (reverse direction) depending on where you want to go.

**Reply by reasonset** [reply]
@JosephMcc @Loufute I didn't know too! Looks rather nice.
